File: Solutions/Hard/1611. Minimum One Bit Operations to Make Integers Zero/Solution.cs
public class Solution {
    public int MinimumOneBitOperations(int n) {
        int result = 0;
        
        while (n > 0) {
            int msb = MSB(n);
            result ^= FlipBits(msb);
            n ^= msb;
        }
        
        return result;
    }

    private int MSB(int n) {
        int msb = 1;
        while (msb <= n) {
            msb <<= 1;
        }
        return msb >> 1;
    }
    
    private int FlipBits(int msb) {
        return (msb << 1) - 1;
    }
}

File: Solutions/Hard/1611. Minimum One Bit Operations to Make Integers Zero/Solution.py
class Solution:
    def minimumOneBitOperations(self, n: int) -> int:
        
        def mostSigBit(n: int) -> int:
            msb = 1
            while(msb <= n):
                msb <<= 1
            return msb >> 1
        
        def flipBits(msb: int) -> int:
            return (msb << 1) - 1

        res: int = 0
        while(n > 0):
            msb: int = mostSigBit(n)
            res ^= flipBits(msb)
            n ^= msb


        return res